### DISCUSSION TITLE
Reset one field resetting the whole form

### DISCUSSION THREAD

**Discussion by mwmcode** [react-hook-form/react-hook-form#5224] [Q&A] [answered]
Hello, 

**Describe the bug**
My understanding is [reset](https://react-hook-form.com/api/useform/reset) can be used to reset one field only (while keeping other dirty ones as they are)

Simple scenario: 2 drop downs (state & city). I want to reset the `city` when the `state` changes

**To Reproduce**

1. Go to [codesandbox](https://codesandbox.io/s/keen-monad-ddoyc?file=/src/App.tsx:524-621)
2. Select state/city works fine
3. Now uncomment `React.useEffect` 
4. Now try to select a state (nothing will be selected cause the `reset` is resetting the whole form)

**Codesandbox link (Required)**
https://codesandbox.io/s/keen-monad-ddoyc?file=/src/App.tsx:524-621

**Expected behavior**
`reset` to only reset the field(s) provided to it

**Screenshots**

https://user-images.githubusercontent.com/13101565/118499582-456df480-b751-11eb-8ef3-959706228473.mov

**Comment by JGibel** [comment, marked as answer]
Why not just call setValue instead?

**Comment by mwmcode** [comment]
I ended up using `setValue` instead. Please close if it's not an issue

**Comment by JGibel** [comment]
csb w/ setValue in case anyone from the internet comes across this in the future: https://codesandbox.io/s/crazy-gates-lkuk8

**Comment by mwmcode** [comment]
@JGibel thanks for the samle.

 
The reason I wanted to do it with `reset` was/is to avoid writing several `setValue`s.

I ended up extracting the logic to a custom hook:

```js
export function useWatchFields() {
  const { watch, setValue } = useFormContext();
  const [region, country, state] = watch(['region', 'country', 'state']);

  useEffect(() => {
    setValue('country', '');
    setValue('state', '');
    setValue('city', '');
  }, [region]);

  useEffect(() => {
    setValue('state', '');
    setValue('city', '');
  }, [country]);

  useEffect(() => {
    setValue('city', '');
  }, [state]);

  return {
    region, country, state,
  };
}

```

**Comment by bluebill1049** [comment]
```
reset({
   ...getValues(),
   // reset goes here
})
```

however, I would still recommend going with setValue, you can easily write an abstraction to do multiple eg setValues

**Reply by mackankowski** [reply]
Hi @bluebill1049 - I'm doing as you mentioned but it cleans nested values - let's see that example:

Before resetting:

![image](https://user-images.githubusercontent.com/9574907/128352541-bb64eaea-540e-4bbe-bb9a-e31d879492b3.png)

After resetting:

![image](https://user-images.githubusercontent.com/9574907/128352895-63c95a01-d668-41d4-b1b3-1bd1fe9cc839.png)

I think the spread operator is the reason because it does a shallow copy instead of a deep one. Any other suggestion, different than using the `setValue()`? Thanks!

---

**UPDATE:** I assume that `setValue('name', 'value', { shouldDirty: true })` works same way like `reset({name: 'value'})` so I can agree on using `setValue()` for my case ;-)
